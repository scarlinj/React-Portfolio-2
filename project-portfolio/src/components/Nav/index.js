import React from 'react';

function Nav() {
  const  categories = [
    { name: 'About Me', description: 'About the developer' },
    { name: 'Portfolio', description: 'A list of projects I developed' },
    { name: 'Resume', description: 'My professional background' },
    { name: 'Contact Me', description: 'Contact me' }
  ];
  // const handleClick = () => {
  //   console.log("click handled")
  // };
  function categorySelected(name) {
    console.log(`${name} clicked`)
  }

    return (
      <header>
        <h1 id="nav"></h1>
        <h2>
          <a href="/">
            <span role="img" aria-label="logo"> üë®‚Äçüíª </span> Stephen Carlin Web Development
          </a>
        </h2>
  <nav>
    <ul className="flex-row">
      {categories.map((category) => (
        <li className="mx-1" key={category.name}>
          {/* onClick() attribute is expecting a callback function declaration.
           It's important that we wrap it in a function declaration rather than just calling categorySelected(category.name), 
           which would cause the function to get called whenever the component renders as well */}
              <span onClick={() => categorySelected(category.name)} >
      {category.name}
    </span>
        </li>
        ))}
    </ul>
  </nav>
      </header>
    );
  }

export default Nav;