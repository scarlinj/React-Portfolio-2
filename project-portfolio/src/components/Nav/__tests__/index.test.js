import React from 'react';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Nav from '..';

afterEach(cleanup);

  describe('Nav component', () => {
    // baseline test
    it('renders', () => {
      render(<Nav />);
    });
    // snapshot test - assert value comparison
      it('matches snapshot', () => {
        const { asFragment } = render(<Nav />);
      
        expect(asFragment()).toMatchSnapshot();
      });
    // emoji test - if emoji is functional
    describe('emoji is visible', () => {
    it('emoji is visible in h2', () => {
        // Arrange
        const { getByLabelText } = render(<Nav />);
        // Assert
        expect(getByLabelText('logo')).toHaveTextContent('👨‍💻');
      });
    })
  })