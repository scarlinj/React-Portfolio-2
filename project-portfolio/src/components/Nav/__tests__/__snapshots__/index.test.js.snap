// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav component matches snapshot 1`] = `
<DocumentFragment>
  <header>
    <h1
      id="nav"
    />
    <h2>
      <a
        href="/"
      >
        <span
          aria-label="logo"
          role="img"
        >
           üë®‚Äçüíª 
        </span>
         Stephen Carlin Web Development
      </a>
    </h2>
    <nav>
      <ul
        class="flex-row"
      >
        <li
          class="mx-1"
        >
          <span>
            About Me
          </span>
        </li>
        <li
          class="mx-1"
        >
          <span>
            Portfolio
          </span>
        </li>
        <li
          class="mx-1"
        >
          <span>
            Resume
          </span>
        </li>
        <li
          class="mx-1"
        >
          <span>
            Contact Me
          </span>
        </li>
      </ul>
    </nav>
  </header>
</DocumentFragment>
`;
